app/assets/src/components/ui/controls/dropdowns/common/DropdownTrigger.jsx

```diff
@@ -50,6 +116,13 @@ DropdownTrigger.propTypes = {
   disabled: PropTypes.bool,
   erred: PropTypes.bool,
   onClick: PropTypes.func,
+
+  /*
+  This prop is needed because we handle filter layout differently on different pages.
```

Comment:
Having looked at the code a bit more, I would suggest adding "This prop should only be used when a DropdownLabel (a.k.a. the "badge") is passed in as the value." To make it more clear that this is meant for DropdownLabels, and the badge refers to the DropdownLabel.